### PR TITLE
Jp/rnns

### PR DESCRIPTION
**Pull request by Henrik-KM**
Added insert command to list, added interlayer dropout to MLP component

### CODE REVIEW

**Review by Henrik-KM — COMMENTED** [review]
Makes it easier to configure dropout in MLP components in a deeplay-like way, i.e. with
MLP.blocks.dropout(p=x)
or
MLP.blocks.insert(index,nn.Module)

**Review by giovannivolpe — APPROVED** [review]

File: deeplay/list.py [hunk 1]
```diff
@@ -37,6 +37,18 @@ def append(self, module: DeeplayModule) -> "LayerList[T]":
...
+        if not self._has_built:
+            args = list(self._args)
+            args.insert(index, module)
+            self._args = tuple(args)
+            self.__construct__()
+
+        else:
+            super().insert(index, module)
+
+        return self
```
Comment (BenjaminMidtvedt) [inline deeplay/list.py:51]: I'm assuming the `@before_build` suggestion didn't work because it didn't apply immediately? We can solve that bycalling before_build callbacks immediately as well. I think that would scale better. Then it would simply be

```py
@before_build
def insert(self, index, module):
    return super().insert(index, module)
```
